### PR TITLE
Nexus: Add unit tests and example for qdens-radial tool

### PR DESCRIPTION
**Pull request by mcbennet**
## Proposed changes

The following PR adds units tests for `qdens-radial` and provides an example of how the tool is used inside `nexus/examples/qmcpack/rsqmc_misc/diamond_radial_density`. This should satisfy #3590 

## What type(s) of changes does this code introduce?

- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

QMCPACK Workshop 2021 Virtual Box

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate) -- in the form of an example

### CODE REVIEW

**Comment by mcbennet** [comment]
Currently, `qdens-radial` depends on Spglib and the tests here will not pass since Spglib is not installed.
Can Spglib be installed on the machines?

File: nexus/tests/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ if(ADD_TEST)
   foreach(TESTNAME ${TESTLIST})
     #message("Adding test ntest_nexus_${TESTNAME}")
     set(NTEST "${qmcpack_SOURCE_DIR}/nexus/bin/nxs-test")
-    add_test(NAME ntest_nexus_${TESTNAME} COMMAND ${NTEST} -R ${TESTNAME} --ctest
```
Comment (jtkrogel) [inline nexus/tests/CMakeLists.txt:16]: What is the added `\$` needed for here?

**Review by jtkrogel — CHANGES_REQUESTED** [review]

File: nexus/tests/CMakeLists.txt (hunk 1 above)
Comment (mcbennet) [reply nexus/tests/CMakeLists.txt:16]: It matches the end of the string. For instance, if `-R qdens` is specified, then qdens and qdens-radial will match, however, if `-R qdens$` is specified then only qdens will match. When there are multiple matches then the tests will not pass -- see for instance cecc6d9

Comment (mcbennet) [reply nexus/tests/CMakeLists.txt:16]: Here is the link to the failed test
https://github.com/QMCPACK/qmcpack/pull/3676/checks?sha=cecc6d9ec198bbf67dcc7886c75d018dd0e66def

Comment (mcbennet) [reply nexus/tests/CMakeLists.txt:16]: Error message placed here for convenience
```
Exactly one test should be selected when using ctest
    tests requested: ['qdens', 'qdens_radial']
```

Comment (jtkrogel) [reply nexus/tests/CMakeLists.txt:16]: Yes, that looks like the behavior we want.  Thanks!

**Comment by jtkrogel** [comment]
@mcbennet In the style of the other Nexus tests, this one should check for existence of spglib via `versions` and exclude the test if it isn't there.  Does this still need to be added?

**Comment by mcbennet** [comment]
Thank you @jtkrogel . The check for Spglib is now implemented

**Review by jtkrogel — DISMISSED** [review]
LGTM

**Review by ye-luo — CHANGES_REQUESTED** [review]
Please remove `p2q.err` and `scf.err` empty files.

File: nexus/tests/unit/test_qdens_radial_files/diamond_twist/dmc/dmc.avg.s001.SpinDensity_u-d.xsf [hunk 2]
```diff
@@ -0,0 +1,35 @@
+ CRYSTAL
```
[the remark this answers is not in the document]
Comment (mcbennet) [reply nexus/tests/unit/test_qdens_radial_files/diamond_twist/dmc/dmc.avg.s001.SpinDensity_u-d.xsf]: I have removed the unnecessary files

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**